db/memorystore.go

```diff
@@ -213,6 +215,19 @@ func (m *MemoryStore) GetAuthorizationByID(id string) *core.Authorization {
 	return m.authorizationsByID[id]
 }
 
+func (m *MemoryStore) FindValidAuthorization(accountID string, identifier acme.Identifier) *core.Authorization {
+	m.RLock()
+	defer m.RUnlock()
+	for _, authz := range m.authorizationsByID {
+		if authz.Identifier.Type == identifier.Type && authz.Identifier.Value == identifier.Value &&
```

Comment:
How do you feel about implementing `func (id acme.Identifier) Equals(other acme.Identifier)` in the `acme `package and using that here?

Reply:
But to be clear I think the `authz.ExpiresDate.After` should remain outside of the `Equals` impl.